include/up-cpp/datamodel/builder/UMessage.h

```diff
@@ -0,0 +1,276 @@
+// limitations under the License.
+//
+// SPDX-FileType: SOURCE
+// SPDX-FileCopyrightText: 2024 Contributors to the Eclipse Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef UP_CPP_DATAMODEL_BUILDER_UMESSAGE_H
+#define UP_CPP_DATAMODEL_BUILDER_UMESSAGE_H
+
+#include <uprotocol/v1/uattributes.pb.h>
+#include <uprotocol/v1/umessage.pb.h>
+#include <uprotocol/v1/upayload.pb.h>
+#include <uprotocol/v1/uri.pb.h>
+#include <uprotocol/v1/uuid.pb.h>
+
+#include <cstdint>
+#include <string>
+#include <vector>
+
+namespace uprotocol::datamodel::builder {
+
+/// @brief Interface for composing UMessage objects
+///
+/// The class provides a set of static methods for constructing standard uP
+/// message types, along with a fluent interface to customize those messages.
+/// A uprotocol::v1::UMessage is produced based on the builder's internal state
+/// when one of the build methods is called.
+///
+/// @remarks For recurring messages (e.g. periodic publishing), the builder
+///          instance can be held and reused by calling .build(payload)
+///          for each new set of message data.
+struct UMessageBuilder {
+	/// @brief Pre-populates a message builder with the attributes of a
+	///        "publish" type message.
+	///
+	/// @param topic UUri the message will be published to.
+	///
+	/// @throws InvalidUUri if URI fails validation.
+	/// @returns UMessageBuilder configured to build a "publish" message
+	static UMessageBuilder publish(v1::UUri&& topic);
+
+	/// @brief Pre-populates a message builder with the attributes of a
+	///        "notification" type message.
+	///
+	/// @param origin UUri the message will be sent from.
+	/// @param destination UUri the message will be sent to.
+	///
+	/// @throws InvalidUUri if URI fails validation.
+	/// @returns UMessageBuilder configured to build a "notification" message
+	static UMessageBuilder notification(v1::UUri&& origin,
+	                                    v1::UUri&& destination);
+
+	/// @brief Pre-populates a message builder with the attributes of an
+	///        RPC "request" type message.
+	///
+	/// @param invoke_method UUri where the RPC request will be serviced.
+	/// @param reply_to UUri where responses will be returned to.
+	/// @param ttl Amount of time from the call to build() that a message
+	///            will still be considered valid. Must be greater than 0
+	///            and fit within a uint32_t.
+	///
+	/// @throws InvalidUUri if URI fails validation.
+	/// @throws std::out_of_range if (ttl <= 0) or (ttl > uint32_t max).
+	/// @returns UMessageBuilder configured to build a "request" message
+	static UMessageBuilder request(v1::UUri&& invoke_method,
+	                               v1::UUri&& reply_to,
+	                               std::chrono::milliseconds ttl);
+
+	/// @brief Pre-populates a message builder with the attributes of an
+	///        RPC "response" type message.
+	///
+	/// @note An alternate responseTo(UMessage) interface is also provided
```

Comment:
```suggestion
	/// @note An alternate response(UMessage) interface is also provided
```

You may have changed the name in the API, but missed updating in the comments?